Applications/spire/source/book_view/book_view_window.cpp

```diff
@@ -44,6 +48,14 @@ BookViewWindow::BookViewWindow(const BookViewProperties& properties,
 }
 
 void BookViewWindow::set_model(std::shared_ptr<BookViewModel> model) {
+  m_transition_widget.reset();
+  m_header_widget->reset_labels();
+  QTimer::singleShot(2000, this, [=] { show_transition_widget(); });
+  m_is_data_loaded = false;
+  m_data_loaded_promise = model->load();
+  m_data_loaded_promise.then(
+    [=] (auto&& value) { on_data_loaded(std::move(value)); });
+  m_model = model;
```

Comment:
When passing parameters by value to take ownership of them, you move the value rather than strictly assign by copy. That is you do this:

m_model = std::move(model);

Then you never reference model again, you strictly refer to m_model.

Furthermore, taking ownership of parameters should be done as soon as possible. That is, move line 58 as close to the function declaration as possible (which in this example means moving it all the way up to line 51).

Exceptions to this rule are when moves and copies perform the exact same action, like with ints, or bools... or other primitive types that do not have a move constructor.